python/plugins/MetaSearch/dialogs/maindialog.py

```diff
@@ -718,35 +718,45 @@ def add_to_ows(self):
 
         caller = self.sender().objectName()
 
-        # stype = human name,/qgis/connections-%s,providername
         if caller == 'mActionAddWms':
-            stype = ['OGC:WMS/OGC:WMTS', 'wms', 'wms']
+            service_type = 'OGC:WMS/OGC:WMTS'
+            sname = 'WMS'
+            dyn_param = ['wms']
+            provider_name = 'wms'
+            setting_node = QgsSettingsTree.node('connections').childNode('ows').childNode('connections')
             data_url = item_data['wms']
         elif caller == 'mActionAddWfs':
-            stype = ['OGC:WFS', 'wfs', 'WFS']
+            service_type = 'OGC:WFS'
+            sname = 'WFS'
+            dyn_param = ['wfs']
+            provider_name = 'WFS'
+            setting_node = QgsSettingsTree.node('connections').childNode('ows').childNode('connections')
             data_url = item_data['wfs']
         elif caller == 'mActionAddWcs':
-            stype = ['OGC:WCS', 'wcs', 'wcs']
+            service_type = 'OGC:WCS'
+            sname = 'WCS'
+            dyn_param = ['wcs']
+            provider_name = 'wcs'
+            setting_node = QgsSettingsTree.node('connections').childNode('ows').childNode('connections')
             data_url = item_data['wcs']
         elif caller == 'mActionAddAms':
-            stype = ['ESRI:ArcGIS:MapServer', 'ams', 'arcgismapserver']
+            # TODO: this doesn't exist anymore?
```

Comment:
Nope, they've all been rolled into the one "arcgis rest services" group

Reply:
thanks